### PR TITLE
request: development 42 api 07 exercise crud set model relationship

### PR DESCRIPTION
**Pull request by OmarAbdiAli**

### CODE REVIEW

File: src/main/java/com/noroff/mefit/controller/SetController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Tag(name = "Set")
+@RequestMapping("/api/set")
+public record SetController(SetService setService) {
+    @GetMapping
+    public List<Set> getAllSets() {
+        return setService.getAll();
+    }
+
+    @PostMapping
+    public Set createSets(@RequestBody Set set) {
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/controller/SetController.java:20]:
```suggestion
    public Set createSet(@RequestBody Set set) {
```

File: src/main/java/com/noroff/mefit/service/SetService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public List<Set> getAll() {
+        return setRepository.findAll();
+    }
+
+    public Set create(Set set) {
+        return setRepository.save(set);
+    }
+
+    public Set getById(Long setId) {
+        if (!setRepository.existsById((setId))) {
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/service/SetService.java:21]: Double parentheses around `setId`
```suggestion
        if (!setRepository.existsById(setId)) {
```

File: src/main/java/com/noroff/mefit/service/SetService.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            return null;
+        }
+
+        return setRepository.findById(setId).orElse(null);
+    }
+
+    public Set update(Set set, Long setId) {
+        if (!setRepository.existsById((setId))) {
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/service/SetService.java:29]:
```suggestion
        if (!setRepository.existsById(setId)) {
```

File: src/main/java/com/noroff/mefit/service/SetService.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            return null;
+        }
+
+        Set prevSet = setRepository.findById(setId).orElse(null);
+        if(prevSet == null) {
+            return null;
+        }
+
+        set.setId(prevSet.getId());
+        //change exercise id as well
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/service/SetService.java:39]:
```java
//change exercise id as well
```
Is this functionality missing?
Or is it a forgotten comment?

**Review by Cusatelli — CHANGES_REQUESTED** [review]
# ❌ Minor changes
Should be changed before merge with dev.

Overall, code looks nice and is working fine, nice job 😄

File: src/main/java/com/noroff/mefit/service/SetService.java (hunk 4 above)
Comment (OmarAbdiAli) [reply src/main/java/com/noroff/mefit/service/SetService.java:39]: Just a forgotten comment. I'll delete that

**Review by Cusatelli — APPROVED** [review]
# ✅ Nice!

**Pull request merged**